# Problem 30
Surprisingly there are only three numbers that can be written as the sum of fourth powers of their digits:

1634 = 14 + 64 + 34 + 44
8208 = 84 + 24 + 04 + 84
9474 = 94 + 44 + 74 + 44
As 1 = 14 is not a sum it is not included.

The sum of these numbers is 1634 + 8208 + 9474 = 19316.

Find the sum of all the numbers that can be written as the sum of fifth powers of their digits.

# Idea:
So, here we are going to repeatedly use the fifith powers of nos from 1-9
so let us create a dictionary holding those fifth powers for the respective keys which range from 1-9.

By doing this we need not calcluate the powers of nos. again and again

# Solution:

In [1]:
d=dict()

In [2]:
for i in range(10):
    d[i]=i**5
d

{0: 0,
 1: 1,
 2: 32,
 3: 243,
 4: 1024,
 5: 3125,
 6: 7776,
 7: 16807,
 8: 32768,
 9: 59049}

In [25]:
powFiveset = set()
s=0
for i in range(2,1000000):
    s=0
    for x in list(str(i)):
        s+=d[int(x)]    
    if(s==i):
        powFiveset.add(i)
        print(i)
        
print(powFiveset)
print(sum(powFiveset))
    

4150
4151
54748
92727
93084
194979
{194979, 4151, 4150, 92727, 54748, 93084}
443839


# Below I'm checking some greater range and getting the same sum 443838 only

* So the answer is 443839.

In [26]:
powFiveset = set()
s=0
for i in range(2,10000000):
    s=0
    for x in list(str(i)):
        s+=d[int(x)]    
    if(s==i):
        powFiveset.add(i)
        print(i)
        
print(powFiveset)
print(sum(powFiveset))

4150
4151
54748
92727
93084
194979
{194979, 4151, 4150, 92727, 54748, 93084}
443839
